### DISCUSSION TITLE
Firmware upgrade to 2.3

### DISCUSSION THREAD

**Discussion by seaky** [seaky/nspanel_pro_tools_apk#32] [General]
[original post](https://github.com/seaky/nspanel_pro_roottool_apk/issues/1#issuecomment-1875847569)

Did you trigger the recovery mode? 
Because it seems you triggered the recovery process which explains why you were reverted to the 1.7

**Comment by bdherouville** [comment]
Yes, when triggering the recovery mode it revert to 1.7 (I thought the update process updates the recovery partitions too.)

But otherwise the upgrades works fine. I can install Home assitant & Linphone without installing Xposed.

**Reply by seaky** [reply]
Yes you dont need Xposed nowadays I’ll write a post about it you can update webview whenever you want easily

**Reply by bdherouville** [reply]
This is the list of the installed packages. It would be great to see what can be removed.

`package:/data/app/com.seaky.nspanelpro.roottools-CxozzSi4qzneGgseK9yNjw==/base.apk=com.seaky.nspanelpro.roottools
package:/system/priv-app/CtsShimPrivPrebuilt/CtsShimPrivPrebuilt.apk=com.android.cts.priv.ctsshim
package:/system/priv-app/TelephonyProvider/TelephonyProvider.apk=com.android.providers.telephony
package:/system/priv-app/CalendarProvider/CalendarProvider.apk=com.android.providers.calendar
package:/system/priv-app/MediaProvider/MediaProvider.apk=com.android.providers.media
package:/system/priv-app/WallpaperCropper/WallpaperCropper.apk=com.android.wallpapercropper
package:/system/priv-app/DocumentsUI/DocumentsUI.apk=com.android.documentsui
package:/vendor/overlay/framework-res__auto_generated_rro.apk=android.auto_generated_rro__
package:/system/priv-app/ExternalStorageProvider/ExternalStorageProvider.apk=com.android.externalstorage
package:/system/app/HTMLViewer/HTMLViewer.apk=com.android.htmlviewer
package:/system/app/CompanionDeviceManager/CompanionDeviceManager.apk=com.android.companiondevicemanager
package:/system/priv-app/MmsService/MmsService.apk=com.android.mms.service
package:/system/priv-app/DownloadProvider/DownloadProvider.apk=com.android.providers.downloads
package:/system/priv-app/DefaultContainerService/DefaultContainerService.apk=com.android.defcontainer
package:/system/app/DownloadProviderUi/DownloadProviderUi.apk=com.android.providers.downloads.ui
package:/system/app/PacProcessor/PacProcessor.apk=com.android.pacprocessor
package:/system/app/CertInstaller/CertInstaller.apk=com.android.certinstaller
package:/system/framework/framework-res.apk=android
package:/system/app/factoryApp/factoryApp.apk=com.eWeLinkNSPro.dev
package:/data/app/io.homeassistant.companion.android-9UL-r3UvCNBcorCl6vmImQ==/base.apk=io.homeassistant.companion.android
package:/system/app/EasterEgg/EasterEgg.apk=com.android.egg
package:/system/priv-app/MtpDocumentsProvider/MtpDocumentsProvider.apk=com.android.mtp
package:/system/app/NfcNci/NfcNci.apk=com.android.nfc
package:/system/priv-app/BackupRestoreConfirmation/BackupRestoreConfirmation.apk=com.android.backupconfirm
package:/data/app/org.linphone-MtM7sp8sQYYUEhnooHBFDQ==/base.apk=org.linphone
package:/system/priv-app/Provision/Provision.apk=com.android.provision
package:/system/priv-app/StatementService/StatementService.apk=com.android.statementservice
package:/vendor/overlay/SysuiDarkTheme/SysuiDarkThemeOverlay.apk=com.android.systemui.theme.dark
package:/vendor/app/RkApkinstaller/RkApkinstaller.apk=com.android.apkinstaller
package:/system/priv-app/SettingsProvider/SettingsProvider.apk=com.android.providers.settings
package:/system/priv-app/SharedStorageBackup/SharedStorageBackup.apk=com.android.sharedstoragebackup
package:/system/app/RkDeviceTest/RkDeviceTest.apk=com.rockchip.devicetest
package:/system/app/PrintSpooler/PrintSpooler.apk=com.android.printspooler
package:/system/app/BasicDreams/BasicDreams.apk=com.android.dreams.basic
package:/system/app/WebView/WebView.apk=com.android.webview
package:/system/app/RkExplorer/RkExplorer.apk=com.android.rk
package:/system/priv-app/InputDevices/InputDevices.apk=com.android.inputdevices
package:/system/app/BuiltInPrintService/BuiltInPrintService.apk=com.android.bips
package:/system/app/kz_app/kz_app.apk=com.eWeLinkControlPanel
package:/data/app/l.l-Igdg6UzWvjXpL1DNqLEjpw==/base.apk=l.l
package:/system/priv-app/MusicFX/MusicFX.apk=com.android.musicfx
package:/data/app/com.google.android.webview-QO6ZfrHsunce9SbiOuys9Q==/base.apk=com.google.android.webview
package:/system/app/ExtShared/ExtShared.apk=android.ext.shared
package:/system/priv-app/OneTimeInitializer/OneTimeInitializer.apk=com.android.onetimeinitializer
package:/system/priv-app/Telecom/Telecom.apk=com.android.server.telecom
package:/system/app/KeyChain/KeyChain.apk=com.android.keychain
package:/system/app/PrintRecommendationService/PrintRecommendationService.apk=com.android.printservice.recommendation
package:/system/priv-app/ExtServices/ExtServices.apk=android.ext.services
package:/system/app/ThomassRoom/ThomassRoom.apk=com.android.gl2jni
package:/system/priv-app/PackageInstaller/PackageInstaller.apk=com.android.packageinstaller
package:/system/app/PicoTts/PicoTts.apk=com.svox.pico
package:/system/priv-app/ProxyHandler/ProxyHandler.apk=com.android.proxyhandler
package:/system/app/LatinIME/LatinIME.apk=com.android.inputmethod.latin
package:/system/app/Browser2/Browser2.apk=org.chromium.webview_shell
package:/system/priv-app/ManagedProvisioning/ManagedProvisioning.apk=com.android.managedprovisioning
package:/system/app/PhotoTable/PhotoTable.apk=com.android.dreams.phototable
package:/data/app/com.seaky.nspanelpro.tools-Bm1KfJXWP6-U-EzR0upGLQ==/base.apk=com.seaky.nspanelpro.tools
package:/system/app/WAPPushManager/WAPPushManager.apk=com.android.smspush
package:/system/app/LiveWallpapersPicker/LiveWallpapersPicker.apk=com.android.wallpaper.livepicker
package:/system/priv-app/StressTest/StressTest.apk=com.cghs.stresstest
package:/system/priv-app/StorageManager/StorageManager.apk=com.android.storagemanager
package:/system/app/BookmarkProvider/BookmarkProvider.apk=com.android.bookmarkprovider
package:/system/priv-app/Settings/Settings.apk=com.android.settings
package:/system/app/Lightning/Lightning.apk=acr.browser.barebones
package:/system/app/CtsShimPrebuilt/CtsShimPrebuilt.apk=com.android.cts.ctsshim
package:/system/priv-app/VpnDialogs/VpnDialogs.apk=com.android.vpndialogs
package:/system/app/Music/Music.apk=com.android.music
package:/system/priv-app/TeleService/TeleService.apk=com.android.phone
package:/system/priv-app/Shell/Shell.apk=com.android.shell
package:/oem/bundled_persist-app/CoolKit_22101317/CoolKit_22101317.apk=com.smatek.test
package:/system/app/WallpaperBackup/WallpaperBackup.apk=com.android.wallpaperbackup
package:/system/priv-app/BlockedNumberProvider/BlockedNumberProvider.apk=com.android.providers.blockednumber
package:/system/app/UserDictionaryProvider/UserDictionaryProvider.apk=com.android.providers.userdictionary
package:/system/priv-app/FusedLocation/FusedLocation.apk=com.android.location.fused
package:/system/priv-app/SystemUI/SystemUI.apk=com.android.systemui
package:/system/app/BluetoothMidiService/BluetoothMidiService.apk=com.android.bluetoothmidiservice
package:/system/priv-app/DeviceTest/DeviceTest.apk=com.DeviceTest
package:/system/app/Bluetooth/Bluetooth.apk=com.android.bluetooth
package:/system/app/WallpaperPicker/WallpaperPicker.apk=com.android.wallpaperpicker
package:/system/priv-app/ContactsProvider/ContactsProvider.apk=com.android.providers.contacts
package:/system/app/CaptivePortalLogin/CaptivePortalLogin.apk=com.android.captiveportallogin
package:/vendor/app/Rk4kVideoPlayer/Rk4kVideoPlayer.apk=android.rk.RockVideoPlayer
`

**Comment by pascalsaul** [comment]
@seaky there is a new firmware 2.4... I had this installed but fucked the device with the not needed xposed apk. After recovery I had 1.7 back again so also not persisted. Maybe someone can get 2.4 and upload to the share? Webview 121 is running fine with Full Kiosk Browser.

**Reply by seaky** [reply]
Latest ota firmware version is v2.3.0  and the latest apk version is 2.4.0.

ota 2.3.0 is available here:
https://drive.google.com/drive/u/2/folders/1bHLLJy8vYUjkCCnyUmrHgEFWi_2cxywv

**Reply by pascalsaul** [reply]
Yeah, I ment ota 2.4.0 is out: https://itead.cc/product/sonoff-nspanel-pro-smart-home-control-panel/
